controller/impl.go

```diff
@@ -73,7 +82,6 @@ func (f *Factory) runCommandLoop(ctx context.Context, logger *zerolog.Logger, po
 	errCh := make(chan error)
 
 	go func() {
```

Comment:
Don't we need an error group here too? No one is waiting for this goroutine to exit.
If you do that you won't need `errCh` anymore because the goroutine could simply return the error.